test/test.js

```diff
@@ -34,9 +36,49 @@ describe('beaver-logger tests', () => {
         });
 
         logEndpoint.expectCalls();
-
         return $logger.flush().then(() => {
             logEndpoint.done();
         });
     });
+
+    it('should log something and flush it to the buffer using sendBeacon', () => {
```

Comment:
Looks good. Can we also get a test to make sure XMLHttpRequest is used if headers are passed?